src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -0,0 +1,78 @@
+        /// <param name="getConnectionStringFromConfigurationFunc">Function to look up the connection string from the configuration</param>
+        /// <param name="getConnectionStringFromSecretFunc">Function to look up the connection string from the secret store</param>
+        /// <param name="options">Options that influence the behavior of the message pump</param>
+        /// <param name="serviceProvider">Collection of services to use</param>
+        public AzureServiceBusMessagePumpSettings(string entityName, string subscriptionName,
+            Func<IConfiguration, string> getConnectionStringFromConfigurationFunc,
+            Func<ISecretProvider, Task<string>> getConnectionStringFromSecretFunc,
+            AzureServiceBusMessagePumpOptions options, IServiceProvider serviceProvider)
+        {
+            Guard.For<ArgumentException>(
+                () => getConnectionStringFromConfigurationFunc == null && getConnectionStringFromSecretFunc == null,
+                "Unable to determine connection string as it was not defined how to look it up");
+            Guard.NotNull(options, nameof(options));
+            Guard.NotNull(serviceProvider, nameof(serviceProvider));
+
+            _serviceProvider = serviceProvider;
+            _getConnectionStringFromConfigurationFunc = getConnectionStringFromConfigurationFunc;
+            _getConnectionStringFromSecretFunc = getConnectionStringFromSecretFunc;
+
+            EntityName = entityName;
+            SubscriptionName = subscriptionName;
+            Options = options;
+        }
+
+        /// <summary>
+        ///     Name of the entity to process
+        /// </summary>
+        /// <remarks>This is optional as the connection string can contain the entity name</remarks>
+        public string EntityName { get; set; }
+
+        /// <summary>
+        ///     Name of the subscription to process
+        /// </summary>
+        /// <remarks>This is only applicable when using Azure Service Bus Topics</remarks>
+        public string SubscriptionName { get; set; }
+
+        /// <summary>
+        ///     Options that influence the behavior of the message pump
+        /// </summary>
+        public AzureServiceBusMessagePumpOptions Options { get; set; }
+
+        /// <summary>
+        ///     Gets the configured connection string
+        /// </summary>
+        /// <returns>Connection string to authenticate with</returns>
+        public async Task<string> GetConnectionStringAsync()
+        {
+            if (_getConnectionStringFromSecretFunc != null)
+            {
+                var secretProvider = _serviceProvider.GetRequiredService<ISecretProvider>();
+                return await _getConnectionStringFromSecretFunc(secretProvider);
+            }
+
+            var configuration = _serviceProvider.GetRequiredService<IConfiguration>();
+            return _getConnectionStringFromConfigurationFunc(configuration);
```

Comment:
Should we add a post-guard to verify if the connection string is not blank?

Reply:
Question is, do we validate it or does Service Bus do that for us?

But I can change it if you want.

Reply:
Since it's a function that we can set, I think we should also guard against the return of the function and not only for `not null`. In more complex scenario's even exception handling could be added, but that's not added-value for this I guess.

Yes, I think a simple guard will do. Good for you?

Reply:
What's the added value of us validating it if the Service Bus client will do this?

I'm not getting the point I think, fail fast would be the only one but then again - What exception do we throw?

Reply:
No, I mean that the user can put whatever code in that function that he/she/x wants; and we're running it without exception handling. All I'm saying. But maybe I, again, am being to paranoid.

Reply:
I think, at least for now, we are good to go. It should simply be a one-liner and if they want to write more advanced stuff, I'd say it's up to them to manage it then, no?

Reply:
Yep.